### PR TITLE
#17 Extract API calls to API Client + Type annotations

### LINKED ISSUE #17: Extract ChainPatrol API calls into a separate file and add type annotations
We're currently just making `axios.post` calls everywhere. It'd be better to have them all in a single file so that we can change things easily. Also, it would be nice to have type annotations for the return types.

### PR DESCRIPTION
**Pull request by MaXeraph**
## Changes
- Add Request, and Response Types annotations. Followed docs:  https://chainpatrol.io/docs/external-api/overview
- Central API Client with URL generation.
- Replaced all existing Axios calls to utilize the API Client

### CODE REVIEW

File: src/commands/setup.ts
Comment (MaXeraph) [inline src/commands/setup.ts:1]: @umar-ahmed Im not certain on how to test this, if you can conduct some QA here to make sure the DiscordGuildStatus is working as expected; that would be appreciated.

File: src/utils/ChainPatrolApiClient.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  attachmentUrls: string[];
+  contactInfo?: string;
+  description: string;
+  discordGuildId?: string;
+  organizationSlug?: string;
+  title: string;
+};
+
+export type ReportCreateOrganizationResponseType = {
+  name: string;
```
Comment (MaXeraph) [inline src/utils/ChainPatrolApiClient.ts:41]: This was inferred from the usage within this repo, [Documentation](https://chainpatrol.io/docs/external-api/report-create#response) is actually missing some info

Comment (umar-ahmed) [reply src/utils/ChainPatrolApiClient.ts:43]: Added the missing fields and created a PR internally to fix docs

**Review by umar-ahmed — APPROVED** [review]
lgtm

**Pull request merged**